src/internal/modules/gauging-stations/controller.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const mappers = require('./lib/mappers');
+
+/**
+ * Main Gauging station page
+ * All data is loaded via shared pre-handlers
+ *
+ * @param {String} request.params.gaugingStationId - gaugingStation guid
+ */
+
+const getLicencesForGaugingStation = async (request, h) => {
+  const { gaugingStationId } = request.params;
+  const { data } = request.pre.gaugingStations;
+  let newData = mappers.mapStationsLicences(data);
+  /* Format data for stations */
+  if (newData.stations) {
+    newData = mappers.mapStations(newData);
+  }
+  /* Format data for tags */
+  if (newData.stations) {
+    newData = mappers.mapTags(newData);
+  }
+  if ((newData.stations.length === 0) || (newData.stations[0].licenceRef === undefined)) {
```

Comment:
If we wish to have a 404 response the returning a Boom.notFound from the handler should work.

However it may be easiest to implement by returning the Boom.notFound from the pre-handler if the station is not found.  If a pre-handler returns a 404 it takes over the HTTP response.

Reply:
loadGaugingStations in pre-handler is returning a Boom.notFound now. The CI scripts is calling above with validation of gaugingstationId.